### ISSUE TITLE
Generated types: Nullable arrays args in queries aren't treated like other nullable scalars

### ISSUE THREAD

**Issue by dhritzkiv** [nestjs/graphql#267]
## I'm submitting a...
<pre><code>
[ ] Regression 
[x] Bug report
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior
When defining a query argument with a nullable array, the resulting generated interface doesn't keep the argument optional, and causes TS compilation error (if this argument comes after non-optional arguments, for example).

For example,

```gql
type Query {
  someNodes(
    # non-nullable array of non-nullable strings
    ids: [String!]!,
    # nullable String
    stringy: String,
    # nullable array of non-nullable strings
    states: [String!]
  ): Boolean
}
```

generates

```ts
export interface IQuery {
    someNodes(ids: string[], stringy?: string, states: string[]): boolean | Promise<boolean>;
}
```

Note `stringy` is optional, whereas `states` is not.

## Expected behavior
The nullable array argument would be marked as optional, like the other arguments that are nullable.

It should generate

```ts
export interface IQuery {
    someNodes(ids: string[], stringy?: string, states?: string[]): boolean | Promise<boolean>;
}
```

## Minimal reproduction of the problem with instructions
https://codesandbox.io/s/nesttypescriptstarter-v1yyj?fontsize=14

See src/generate-types, src/graphql, src/graphql.ts

## Environment

<pre><code>
Nest version: 6.2.1

Not sure if this is related to this package, or something upstream.

**Comment by Fyzu** [comment]
Same situation

**Comment by kamilmysliwiec** [comment]
Published as 6.2.3
